src/H5Ocopy.c

```diff
@@ -1464,22 +1464,28 @@ H5O__copy_search_comm_dt(H5F_t *file_src, H5O_t *oh_src, H5O_loc_t *oloc_dst /*i
 
             /* Walk through the list of datatype suggestions */
             while (suggestion) {
+                bool exists = false;
+
                 /* Find the object */
-                if (H5G_loc_find(&dst_root_loc, suggestion->path, &obj_loc /*out*/) < 0)
-                    /* Ignore errors - i.e. suggestions not present in
-                     * destination file */
-                    H5E_clear_stack();
```

Comment:
If you haven't already, it's worth documenting that developers should avoid calling `H5E_clear_stack()` in the library since I can easily see this cropping up again.

Reply:
Good point, I'll add that in the next PR.